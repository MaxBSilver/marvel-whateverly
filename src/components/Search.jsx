import React, { Component } from 'react'
export class Search extends Component {
  constructor(props) {
    super(props);
    this.state = {
      value: ''
    }
  }

  handleChange = e => {
    this.setState({ value: e.target.value })
  }

  handleSubmit = e => {
    e.preventDefault()
    const specChars = ["'", " ", "-", ",", ":", "#", ".", "!", "?"]
    let userInput = this.state.value.toUpperCase().split('').filter((ltr) => {
      return !specChars.includes(ltr)
    }).join('')
    console.log(userInput)
    let a = this.props.data.filter((item) => {
      return userInput.match(item.title.toUpperCase().split('').filter((ltr) => {
        return !specChars.includes(ltr)
      }).join('')) ?
      item : false
    })
    this.props.storeRendered(a);
  }

  render() {
    return (
      <form onSubmit={this.handleSubmit}>
        <input
          type='text'
          value={this.state.value}
          onChange={this.handleChange}
          placeholder='Search'
        />
        <input
          type='submit'
          value='Submit'
        />
      </form>
    )
  }
}

export default Search
