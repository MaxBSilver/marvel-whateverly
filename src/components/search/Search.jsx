import React, { Component } from 'react'
export class Search extends Component {
  constructor(props) {
    super(props);
    this.state = {
      value: ''
    }
  }

  handleChange = e => {
    this.setState({ value: e.target.value })
  }

  rmSpecChars(word) {
    return word.split('').filter((ltr) => {
      return !["'", " ", "-", ",", ":", "#", ".", "!", "?"].includes(ltr)
    }).join('')
  }

  finder(words) {
    return this.props.data.filter((item) => { 
      let reduced = item.title.toUpperCase().split(' ').reduce((acc, word) => {
          acc.push(this.rmSpecChars(word))
        return acc;
      }, [])
      return reduced.some(keyWord => words.includes(keyWord));
   })
  }

  handleSubmit = e => {
    e.preventDefault()
    let userInput = this.state.value.split(' ').length > 1 
      ? this.state.value.toUpperCase().split(' ').map(word => this.rmSpecChars(word))
      : [this.rmSpecChars(this.state.value.toUpperCase())]
    let result = this.finder(userInput);
    this.props.storeRendered(result)
  }

  render() {
    return (
      <form onSubmit={this.handleSubmit}>
        <input
          type='text'
          value={this.state.value}
          onChange={this.handleChange}
          placeholder='Search'
        />
        <input
          type='submit'
          value='Submit'
        />
      </form>
    )
  }
}

export default Search
