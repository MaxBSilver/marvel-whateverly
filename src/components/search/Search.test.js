import React from 'react';
import ReactDOM from 'react-dom';
import Search from './Search';
import { shallow } from 'enzyme';

describe('Search', () => {

  const mockStoreRendered = jest.fn()

  const combined = [];

  let wrapper;

  beforeEach(() => {
    wrapper = shallow (
      <Search rendered={combined} storeRendered={mockStoreRendered} />
    )
  });

  it('should have a basic default state', () => {
    expect(wrapper.state()).toEqual( {value: ''} )
  });

  it('should match the snapshot with all the data passed in', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('sould remove special characters', () => {
    expect(wrapper)
  })
})