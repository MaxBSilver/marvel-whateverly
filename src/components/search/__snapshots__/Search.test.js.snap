// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search should match the snapshot with all the data passed in 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    placeholder="Search"
    type="text"
    value=""
  />
  <input
    type="submit"
    value="Submit"
  />
</form>
`;
